Add new Required<T, K> as the opposite of Optional<T, K>

spencersutton opened the pull request:
<!-- Thank you for your contribution! :thumbsup: -->
<!-- Please makes sure that these checkboxes are checked before submitting your PR, thank you! -->

## Description
Adds a new type Required<T, K> that makes keys K in type T required (default keyof T). Compatible with the built-in Required type.

Credit to @vhenzl for the basic implementation.

## Related issues:
- Resolved #93

## Checklist

* [x] I have read [CONTRIBUTING.md](../CONTRIBUTING.md)
* [x] I have linked all related issues above
* [ ] I have rebased my branch

For new features:
* [x] I have added entry in TOC and API Docs
* [x] I have added a short example in API Docs to demonstrate new usage
* [x] I have added type unit tests with `dts-jest`

piotrwitek commented on src/mapped-types.ts:
```diff
@@ -614,3 +614,26 @@ export type ValuesType<
+  K
+> &
+  Required<Pick<T, K>>;
+export { AugmentedRequired as Required };
```
Could you please use normal export on AugmentedRequired here in this module for clearer internal references, and move named export to the API exports in the main index.ts?

spencersutton commented:
I did miss adding it to index.ts. I've added that and incorporated your feedback. AugmentedRequired is now used everywhere and is renamed Required in the index.ts.

piotrwitek commented:
I'll prepare a release tomorrow.
